merkle/src/merkle.rs

```diff
@@ -211,6 +238,17 @@ impl<E: Element> Store<E> for MmapStore<E> {
         self.len += 1;
     }
 
+    fn write_range(&mut self, buf: &[u8], start: usize) {
+        let b = E::byte_len();
+        assert_eq!(buf.len() % b, 0);
+        let r = std::ops::Range {
+            start: start * b,
+            end: start * b + buf.len(),
+        };
+        self.store[r].copy_from_slice(buf);
+        self.len += buf.len() / b;
```

Comment:
Hmm, this seems wrong in some cases, as this could write over existing data, instead of actually extending things. (I think this is a similar issue, as you mention in the impl for mmap)

Could we change the method to be `extend_from_slice(buf: &[u8])` instead and always append to the end?

Reply:
agreed, I was following the pattern from `write_at` which I think is also wrong, so I'll fix both